compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -3553,26 +3563,139 @@ private STNode parseVarDeclRhs(STNode metadata, STNode finalKeyword, STNode type
+                break;
+            default:
+                typeDescriptor = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(typeDescriptor, isolatedQualifier);
+        }
+
+        return STNodeFactory.createTypedBindingPatternNode(typeDescriptor, bindingPattern);
+    }
+
+    /**
+     * Add isolated qualifier to object type descriptor and modify the node.
+     *
+     * @param objectTypeDesc  Object type descriptor node to be modified
+     * @param isolatedKeyword Isolated keyword to be made an object type qualifier
+     * @return Modified node
+     */
+    private STNode modifyObjectTypeDescWithIsolatedQualifier(STNode objectTypeDesc, STNode isolatedKeyword) {
+        List<STNode> newQualifierList = new ArrayList<>();
+        newQualifierList.add(isolatedKeyword);
+
+        STObjectTypeDescriptorNode objectTypeDescriptorNode = (STObjectTypeDescriptorNode) objectTypeDesc;
+        STNodeList qualifierList = (STNodeList) objectTypeDescriptorNode.objectTypeQualifiers;
+
+        for (int i = 0; i < qualifierList.size(); i++) {
+            STNode qualifier = qualifierList.get(i);
+            if (qualifier.kind == SyntaxKind.ISOLATED_KEYWORD) {
+                updateLastNodeInListWithInvalidNode(newQualifierList, qualifier,
+                        DiagnosticErrorCode.ERROR_DUPLICATE_QUALIFIER, ((STToken) qualifier).text());
+            } else {
+                newQualifierList.add(qualifier);
+            }
+        }
+
+        STNode newObjectTypeQualifiers = STNodeFactory.createNodeList(newQualifierList);
+        return objectTypeDescriptorNode.modify(newObjectTypeQualifiers, objectTypeDescriptorNode.objectKeyword,
+                objectTypeDescriptorNode.openBrace, objectTypeDescriptorNode.members,
+                objectTypeDescriptorNode.closeBrace);
+    }
+
+    /**
+     * Add isolated qualifier to function type descriptor and modify the node.
+     *
+     * @param funcTypeDesc  Function type descriptor node to be modified
+     * @param isolatedKeyword Isolated keyword to be made an object type qualifier
+     * @return Modified node
+     */
+    private STNode modifyFuncTypeDescWithIsolatedQualifier(STNode funcTypeDesc, STNode isolatedKeyword) {
+        List<STNode> newQualifierList = new ArrayList<>();
+        newQualifierList.add(isolatedKeyword);
+
+        STFunctionTypeDescriptorNode funcTypeDescriptorNode = (STFunctionTypeDescriptorNode) funcTypeDesc;
+        STNodeList qualifierList = (STNodeList) funcTypeDescriptorNode.qualifierList;
+
+        for (int i = 0; i < qualifierList.size(); i++) {
+            STNode qualifier = qualifierList.get(i);
+            if (qualifier.kind == SyntaxKind.ISOLATED_KEYWORD) {
+                updateLastNodeInListWithInvalidNode(newQualifierList, qualifier,
+                        DiagnosticErrorCode.ERROR_DUPLICATE_QUALIFIER, ((STToken) qualifier).text());
+            } else {
+                newQualifierList.add(qualifier);
+            }
+        }
```

Comment:
Qualifier extraction part is common to this and the method above this one. Can extract to a common method.